ledfx/devices/launchpad.py

```diff
@@ -0,0 +1,149 @@
+# import logging
+
+import launchpad_py as launchpad
+import serial
+import voluptuous as vol
+
+from ledfx.devices import LaunchpadDevice, packets
+
+# _LOGGER = logging.getLogger(__name__)
+
+
+class LaunchpadDevice(LaunchpadDevice):
+    """Launchpad device support"""
+
+    @staticmethod
+    @property
+    def CONFIG_SCHEMA():
+        return vol.Schema(
+            {
+                vol.Required(
+                    "pixel_count",
+                    description="Number of individual pixels",
+                    default=1,
+                ): vol.All(int, vol.Range(min=1)),
+                vol.Required(
+                    "rows",
+                    description="Number of individual rows",
+                    default=1,
+                ): vol.All(int, vol.Range(min=1)),
+            }
+        )
+
+    def __init__(self, ledfx, config):
+        super().__init__(ledfx, config)
+        # try:
+        # 	import launchpad_py as launchpad
+        # except ImportError:
+        # 	try:
+        # 		import launchpad
+        # 	except ImportError:
+        # 		sys.exit("ERROR: loading launchpad.py failed")
+
+        # # create an instance
+        # lp = launchpad.Launchpad()
+        self.lp = launchpad.Launchpad()
+
+        # Clear the buffer because the Launchpad remembers everything
+        # self.lp.ButtonFlush()
+
+        # List the class's methods
+        print(" - Available methods:")
+        for mName in sorted(dir(self.lp)):
+            if mName.find("__") >= 0:
+                continue
+            if callable(getattr(self.lp, mName)):
+                print("     " + str(mName) + "()")
```

Comment:
Move all of this into a dump_methods( lp_instance ) function. Its diagnostic and best stand separate.

Then you can put the call to it under a debug test similar to 

https://github.com/LedFx/LedFx/blob/main/ledfx/__main__.py#L389

Remember to close the lp instance ( doesn't need to be self in this case ) as it is only being used to dump diag here, and gets replaced when activate is called.

Move the prints to _LOGGER.debug(...)

Use f-strings, your brain will thank you https://realpython.com/python-f-strings/